### PR TITLE
readme refactor

### PR DESCRIPTION
**Pull request by jspaaks**
Refactored the README a bit. Some snippets ended up being not used, they are included at the bottom of the document but have been commented out with HTML comments.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,44 +1,38 @@
...
+## Example program: Newton-Raphson root finding
 
-Let's first define the mathematical equation, which we will be searching for its root, and the derivative of it.
+The [Newton-Raphson root finding algorithm](https://en.wikipedia.org/wiki/Newton%27s_method) will be the use case. The
+algorithm is explained in [this video series](https://www.youtube.com/watch?v=cOmAk82cr9M). The code we are using came
+from [geeksforgeeks.org](https://www.geeksforgeeks.org/program-for-newton-raphson-method/).
 
-The equation and its derivative which we will use in this guide are $x^3 - x^2  + 2$ and $3x^2 - 2x$ respectively.
-The root is the value (X-coordinate) which makes the mathematical function (Y-coordinate) equal to `0`. In this equation the root is `-1`.
+Let's first define the mathematical function and its derivative, which we need in order to find the root.
```
Comment (sverhoeven) [inline README.md:31]: Why did you rename equation back to function?

Comment (jspaaks) [reply README.md:31]: I didn't realize that was what it said before. I renamed it because it more precisely captures the meaning of the dollared content. Also, 'derivative' makes more sense for a function than for an equation.

**Comment by jspaaks** [comment]
merge target is now ``develop``.
